src/runtime/write_debug_image.cpp

```diff
@@ -216,24 +219,27 @@ WEAK extern "C" int32_t halide_debug_to_file(void *user_context, const char *fil
             int32_t offset = sizeof(header) + channels * sizeof(int32_t) * 2;
 
             for (int32_t i = 0; i < channels; i++) {
-                if (!f.write((void*)(&offset), 4)) {
+                if (!f.write((void *)(&offset), 4)) {
                     return -4;
                 }
                 offset += shape[0].extent * shape[1].extent * depth * bytes_per_element;
             }
             int32_t count = shape[0].extent * shape[1].extent * depth * bytes_per_element;
             for (int32_t i = 0; i < channels; i++) {
-                if (!f.write((void*)(&count), 4)) {
+                if (!f.write((void *)(&count), 4)) {
                     return -5;
                 }
             }
         }
     } else if (ends_with(filename, ".mat")) {
         // Construct a name for the array from the filename
         const char *start, *end;
-        for (end = filename; *end; end++);
-        for (; *end != '.'; end--);
-        for (start = end; start != filename && start[-1] != '/'; start--);
+        for (end = filename; *end; end++)
```

Comment:
This pattern again... maybe see what it looks like as a while loop?

Reply:
done